src/crimson/osd/replicated_recovery_backend.cc

```diff
@@ -401,7 +401,7 @@ ReplicatedRecoveryBackend::prep_push(
   push_info.recovery_info.ss = push_info_ss;
   push_info.recovery_info.soid = soid;
   push_info.recovery_info.oi = obc->obs.oi;
-  push_info.recovery_info.version = obc->obs.oi.version;
+  push_info.recovery_info.version = need;
```

Comment:
On the replica (or primary) actually doing the push, these should be the same as we do the push from a replica (or primary) with the version we need.  Need might be slightly better because of the mark_unfound_lost_revert behavior, but it shouldn't have mattered to the test case in the commit message.

The commit message links directly to the teuthology failure.  Instead, file a tracker ticket and link to that for two reasons: the teuthology output will be removed within a matter of weeks and it's not clear (to me, at least) how this change fixes that failure.

Reply:
It looks like `need` is also wrong (12) in the test case that failed:
https://gist.github.com/Matan-B/02e88c375760c1cb29f87d39de9122c9

CC: @xuxuehan 

>Need might be slightly better because of the mark_unfound_lost_revert behavior, but it shouldn't have mattered to the test case in the commit message.

`mark_unfound_lost (LOST_REVERT)` is not yet implemented. Perhaps this is why it didn't matter until now.

Reply:
Followup:
I suspect the following missing bit in Crimson is the cause (from `PG::log_operation`):
```
    auto last = logv.rbegin();
    if (is_primary() && last != logv.rend()) {
      projected_log.skip_can_rollback_to_to_head();
      projected_log.trim(cct, last->version, nullptr, nullptr, nullptr);
    }
```

Reply:
Found the cause:
https://tracker.ceph.com/issues/65491